{lib}[GCCcore/12.3.0] PyTorch-bundle v2.1.2, SoX v14.4.2, libmad v0.15.1b, ... w/ CUDA 12.1.1

Micket opened the pull request:
(created using `eb --new-pr`)

I have combined what I found from several conflicting PRs here to make what I think we want to have as the PyTorch-bundle. 

I made some choices here:

1. Stole some from other PRs to combine into this one
2. Using SentencePiece with the existing gperftools instead of introducing a duplicate
3. Verified torchvision and torchtext was using supported versions for PyTorch 2.1
4. Added torchaudio
5. kept torchdata despite it being abandoned (since it was only just abandoned relatively recently)

(I don't really mind how this gets merged, if someone wants to update their PR instead, feel free to close this or just use it as a reference).

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
gpu22.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/0c24f489ee602ccdae65b2561aa04572 for a full test report.

jfgrimm commented:
@boegelbot: please test @ jsc-zen3

Micket commented:
> RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx

huh, jsc-zen3 should have that right?

jfgrimm commented:
@boegelbot: please test @ jsc-zen3
EB_ARGS=--include-easyblocks-from-pr=3236

SebastianAchilles commented:
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
jsczen3g1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.3, x86_64, AMD EPYC-Milan Processor (zen3), 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.23.08, Python 3.9.18
See https://gist.github.com/SebastianAchilles/3d75091724ac38810a1d1d29c8ff4744 for a full test report.

jfgrimm commented:
@boegelbot: please test @ generoso
EB_ARGS=--include-easyblocks-from-pr=3236

jfgrimm commented:
@Micket I've made a PR to your fork that makes sure the easybuild provided sox is used for torchaudio: https://github.com/Micket/easybuild-easyconfigs/pull/19

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
gpu21.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/de8c44168aed72593159b5fbc6822460 for a full test report.

jfgrimm commented:
Test report by @jfgrimm
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3236
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
himem01.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/9887ac847fe7ff2acdcfe08bad3d9822 for a full test report.

jfgrimm commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,92 @@
+    ('torchaudio', '2.2.1', {
+        'installopts': '--no-use-pep517 -v',
+        'patches': ['torchaudio-2.2.1_use-external-sox.patch'],
+        'preinstallopts': 'rm -r third_party/{sox,ffmpeg/multi} && USE_CUDA=1 USE_OPENMP=1 USE_FFMPEG=1 FFMPEG_ROOT="$EBROOTFFMPEG"',
```
```suggestion
        'preinstallopts': ('rm -r third_party/{sox,ffmpeg/multi} &&'
                           'USE_CUDA=1 USE_OPENMP=1 USE_FFMPEG=1 FFMPEG_ROOT="$EBROOTFFMPEG"'),
```

Micket commented:
Test report by @Micket
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
vera-gpu1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 545.23.08, Python 3.6.8
See https://gist.github.com/Micket/a48da3c60257a5558e05e968a779a5f8 for a full test report.

jfgrimm commented:
@boegelbot: please test @ generoso
EB_ARGS=--include-easyblocks-from-pr=3236

jfgrimm commented:
@boegelbot: please test @ jsc-zen3
EB_ARGS=--include-easyblocks-from-pr=3236

jfgrimm reviewed (approved):
lgtm

migueldiascosta commented:
I'm getting this on Arm (Grace-Hopper):
```
== 2024-03-01 10:17:48,455 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/ebinstall/develop/easybuild-framework/easybuild/base/exceptions.py:126 in __init__): Sanity check failed: extensions sanity check failed for 1 extensions: torchtext
failing sanity check for 'torchtext' extension: command "python -c "import torchtext"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/apps/easybuild/software/PyTorch-bundle/2.1.2-foss-2023a-CUDA-12.1.1/lib/python3.11/site-packages/torchtext/__init__.py", line 6, in <module>
    from torchtext import _extension  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/apps/easybuild/software/PyTorch-bundle/2.1.2-foss-2023a-CUDA-12.1.1/lib/python3.11/site-packages/torchtext/_extension.py", line 64, in <module>
    _init_extension()
  File "/apps/easybuild/software/PyTorch-bundle/2.1.2-foss-2023a-CUDA-12.1.1/lib/python3.11/site-packages/torchtext/_extension.py", line 58, in _init_extension
    _load_lib("libtorchtext")
  File "/apps/easybuild/software/PyTorch-bundle/2.1.2-foss-2023a-CUDA-12.1.1/lib/python3.11/site-packages/torchtext/_extension.py", line 50, in _load_lib
    torch.ops.load_library(path)
  File "/apps/easybuild/software/PyTorch/2.1.2-foss-2023a-CUDA-12.1.1/lib/python3.11/site-packages/torch/_ops.py", line 852, in load_library
    ctypes.CDLL(path)
  File "/apps/easybuild/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: /apps/easybuild/software/gperftools/2.12-GCCcore-12.3.0/lib/libtcmalloc_minimal.so.4: cannot allocate memory in static TLS block,  (at easybuild/ebinstall/develop/easybuild-framework/easybuild/framework/easyblock.py:3669 in _sanity_check_step)
```

seems to be related to https://bugzilla.redhat.com/show_bug.cgi?id=1722181#c6, and ld_preloading libtcmalloc_minimal.so.4 does work, not sure if there's anything else we can do about it...

Micket commented:
@migueldiascosta hm, tricky. tcmalloc is even pulled in as a indirect dependency here so i wasn't even expected it to affect anything here. 

I also realized i didn't run any tests, and I think it's definitely worth fixing that, though that will likely just reveal more issues and not fix anything.

Micket commented:
Test report by @Micket
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
vera-gpu1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 545.23.08, Python 3.6.8
See https://gist.github.com/Micket/b47f0c6e7394d5ed49bafb2e0d0888e8 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded (with --ignore-test-failure) for 17 out of 25 (4 easyconfigs in total)
alvis6-09 - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 1 x NVIDIA NVIDIA A40, 535.104.05, Python 3.6.8
See https://gist.github.com/VRehnberg/0ca33036d57ac003b63de0feeed275ca for a full test report.

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded (with --ignore-test-failure) for 4 out of 4 (4 easyconfigs in total)
gpu13.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, 1 x NVIDIA NVIDIA A40, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/2457eed2287959bdf4189f2d65cc9631 for a full test report.

jfgrimm commented:
Test report by @jfgrimm
**FAILED**
Build succeeded (with --ignore-test-failure) for 3 out of 4 (4 easyconfigs in total)
gpu13.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, 1 x NVIDIA NVIDIA A40, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/5d5a225aa24ba9fd107fd841593685ba for a full test report.

jfgrimm commented:
Test report by @jfgrimm
**FAILED**
Build succeeded (with --ignore-test-failure) for 3 out of 4 (4 easyconfigs in total)
gpu06.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7643 48-Core Processor, 1 x NVIDIA NVIDIA A40, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/8d82c2403884b861b337d28c57dfb67b for a full test report.

VRehnberg commented:
## torchtext
Tests are mostly fine. Running torchaudio_unittests on a machine without GPUs, these are the errors I got:
```
FAILED test/torchtext_unittest/test_build.py::TestDataUtils::test_get_tokenizer_moses - ModuleNotFoundError: No module named 'sacremoses'                                                                                                                     
FAILED test/torchtext_unittest/test_build.py::TestDataUtils::test_get_tokenizer_spacy - ModuleNotFoundError: No module named 'spacy'                                                                                                                          
FAILED test/torchtext_unittest/test_build.py::TestVocab::test_download_glove_vectors - EOFError: Ran out of input
FAILED test/torchtext_unittest/test_build.py::TestVocab::test_vectors_get_vecs - EOFError: Ran out of input
```

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,104 @@
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('RE2', '2023-08-01'),  # for torchtext
+    ('parameterized', '0.9.0'),
```
```suggestion
    ('parameterized', '0.9.0'),
    ('scikit-learn', '1.3.1'),
    ('matplotlib', '3.7.2'),
```
sklearn is used a bit here and there. Matplotlib I think it was in pytorch-ignite: tests/ignite/engine

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,104 @@
+            {'torchtext-0.14.1_use-system-libs.patch':
+             '1366d10c4755b6003194f7313ca11d165a80a13d325bee9d669ea2b333d82536'},
+        ],
+        'runtest': 'pytest',
```
```suggestion
        'runtest': 'pytest test/torchtext_unittest',
```
this exludes `test/integration_tests` which I see as a good thing

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,104 @@
+            {'torchaudio-2.2.1_use-external-sox.patch':
+             'fc97b2bdf0ab16b9c4706bf2a31dd72d0ecb09e4af2f9c4a34e18fca8e0d165d'},
+        ],
+        'runtest': 'pytest',
```
```suggestion
        'runtest': (
            'pytest'  # ignored tests require librosa
            ' --ignore=test/torchaudio_unittest/functional/librosa_compatibility_cpu_test.py'
            ' --ignore=test/torchaudio_unittest/functional/librosa_compatibility_cuda_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/functional/librosa_compatibility_cpu_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/functional/librosa_compatibility_cuda_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/transforms/librosa_compatibility_cpu_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/transforms/librosa_compatibility_cuda_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/transforms/librosa_compatibility_cpu_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/transforms/librosa_compatibility_cuda_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/hifi_gan/hifi_gan_cpu_test.py'
            ' --ignore=test/torchaudio_unittest/prototype/hifi_gan/hifi_gan_gpu_test.py'
            ' test/torchaudio_unittest/'
        ),
```
It's either this or pulling in librosa as a build dependency (which would be a better solution, but then someone has to create that EC first)

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,104 @@
+        'source_urls': ['https://github.com/pytorch/ignite/archive'],
+        'sources': [{'download_filename': 'v%(version)s.tar.gz', 'filename': '%(name)s-%(version)s.tar.gz'}],
+        'checksums': ['bfe4b6f1cd96e78c021a65a0c51350cdb89d6ef5a8b9609638666ca95bae51d7'],
+        'runtest': 'pytest',
```
```suggestion
        'runtest': 'pytest --ignore=tests/ignite/contrib --ignore=/tests/ignite/metrics',
        'testinstall': True,
```
Contrib I would think from the name would be perfectly fine to skip for our purposes. Might want to skip tests/ignite/metrics as well or pull in:
 - dill
 - skimage
 - pytorch_fid
 - nltk
 as builddependencies

stevenvdb commented:
A few (5) torchaudio tests are failing when building on a zen4 architecture with H100 GPUs:
```
E       RuntimeError: Failed to create the filter from "scale_cuda=format=yuv444p" (Filter not found.)
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/io/stream_reader_test.py -k test_scale_cuda_format
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```
```
E       RuntimeError: Failed to create the filter from "scale_cuda=iw/2:ih/2" (Invalid argument.)
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/io/stream_reader_test.py -k test_sclae_cuda_change_size
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```
```
E       AssertionError: Tensor-likes are not close!
E
E       Mismatched elements: 2 / 96000 (0.0%)
E       Greatest absolute difference: 2.1187588572502136e-08 at index (1, 1, 15856) (up to 1e-08 allowed)
E       Greatest relative difference: 0.00015638672630302608 at index (1, 1, 15926) (up to 1e-05 allowed)
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/transforms/batch_consistency_test.py -k test_batch_inverse_spectrogram
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```
```
E       AssertionError: Tensor-likes are not close!
E
E       Mismatched elements: 357 / 2400 (14.9%)
E       Greatest absolute difference: 7.092952728271484e-06 at index (1, 0, 264) (up to 1e-08 allowed)
E       Greatest relative difference: 0.004654205869883299 at index (1, 0, 360) (up to 1e-05 allowed)
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/transforms/batch_consistency_test.py -k test_batch_pitch_shift
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```
```
E       AssertionError: Tensor-likes are not close!
E
E       Mismatched elements: 1 / 49446 (0.0%)
E       Greatest absolute difference: 5.3085386753082275e-08 at index (0, 1, 187, 40) (up to 1e-08 allowed)
E       Greatest relative difference: 2.1108940927661024e-05 at index (0, 1, 187, 40) (up to 1e-05 allowed)
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/transforms/batch_consistency_test.py -k test_batch_spectrogram
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```

I don't have enough affinity with this software to judge whether these tests can be ignored or if they indicate an actual problem with the build.

Test report: 
[easybuild-PyTorch-bundle-2.1.2-20240320.163958.CvSim_test_report.md](https://github.com/easybuilders/easybuild-easyconfigs/files/14692973/easybuild-PyTorch-bundle-2.1.2-20240320.163958.CvSim_test_report.md)

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
alvis1-02 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/905adf0f6e076a140fee3317a207850c for a full test report.

Micket commented:
@stevenvdb the FFmpeg tests with the filters, make sure you ahve a ffmpeg built with nvidia drivers and all present (just a guess, but this seems to be isolated to ffmpeg missing the "scale_cuda" part completely, not actually an issue with the torchaudio code).

I'm gonna have a last go at those failing precision tests, upping the tolerances.

yqshao commented:
Not sure if it's relevant for the test tolorences: torchaudio was wrapping transformations in `_DeterministicWrapper` in their [autograd_test](https://github.com/pytorch/audio/blob/release/2.2/test/torchaudio_unittest/transforms/autograd_test_impl.py), but not in e.g. batch_consistency. There is also [torch.use_deterministic_algorithms](https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms)
 
> ```
> E       AssertionError: Tensor-likes are not close!
> E
> E       Mismatched elements: 1 / 49446 (0.0%)
> E       Greatest absolute difference: 5.3085386753082275e-08 at index (0, 1, 187, 40) (up to 1e-08 allowed)
> E       Greatest relative difference: 2.1108940927661024e-05 at index (0, 1, 187, 40) (up to 1e-05 allowed)
> E
> E       To execute this test, run the following from the base repo dir:
> E            python test/torchaudio_unittest/transforms/batch_consistency_test.py -k test_batch_spectrogram
> E
> E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
> ```

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded (with --ignore-test-failure) for 4 out of 4 (4 easyconfigs in total)
alvis3-23 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 550.54.15, Python 3.6.8
See https://gist.github.com/VRehnberg/38c0ec9c8bfa6084d5889e0611ee101a for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded (with --ignore-test-failure) for 4 out of 4 (4 easyconfigs in total)
alvis1-02 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/0ff5e4fd4677e42da67f6d097f6bcf18 for a full test report.

VRehnberg commented:
> FAILED test/test_profiler.py::TestProfiler::test_dump_gpu_metrics - FileNotFoundError: [Errno 2] No such file or directory: '/dev/shm/PyTorchbundle/2.1.2/foss-2023a-CUDA-12.1.1/torchtbprofiler/torch_tb_profiler-0.4.3/test/gpu_metrics_input.json'

There are three files missing under `torch_tb_profiler-0.4.3/test/` in the PyPI archive compared to github https://github.com/pytorch/kineto/tree/8aa6ffbe8762173a617fc18e8d11f263237e586d/tb_plugin/test The PyPI archive only got the python files from `test/`

Micket commented:
FFmpeg filters requires nvcc at compile time, no way to supply that as extra filters. Giving up on it.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
alvis3-13 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 2 x NVIDIA NVIDIA A100-SXM4-40GB, 550.54.15, Python 3.6.8
See https://gist.github.com/VRehnberg/8f82d7243c91492324423c7c46ea8827 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
alvis1-05 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 2 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/a8486d88625f64a42a459b74cf204ad9 for a full test report.

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,159 @@
+        ],
+        'runtest': (
+            'pytest test/torchaudio_unittest/'
+            '-k "not TestProcessPoolExecutor"'  # hang maybe related https://github.com/pytorch/audio/issues/1021
```
```suggestion
            ' -k "not TestProcessPoolExecutor"'  # hang maybe related https://github.com/pytorch/audio/issues/1021
```

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,159 @@
+            'pytest'
+            ' --ignore=tests/ignite/contrib/handlers/test_clearml_logger.py'  # requires clearml
+            ' --ignore=tests/ignite/metrics/gan/test_fid.py'  # requires pytorch_fid
+            '-k "not test__native_dist_model_create_no_dist_nccl"'  # hangs
```
```suggestion
            ' -k "not test__native_dist_model_create_no_dist_nccl"'  # hangs
```

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,159 @@
+        'runtest': (
+            'pytest'
+            ' --ignore=test/test_tensorboard_end2end.py'  # timeouts
+            '-k "not test_dump_gpu_metrics"'  # missing file
```
```suggestion
            ' -k "not test_dump_gpu_metrics"'  # missing file
```

VRehnberg reviewed (commented):
Missing spaces in `runtest` commands.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
alvis1-10 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 2 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/83333c505b3b469a32e67eb0519e5f6b for a full test report.

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,159 @@
+        'runtest': (
+            'pytest'
+            ' --ignore=tests/ignite/contrib/handlers/test_clearml_logger.py'  # requires clearml
+            ' --ignore=tests/ignite/metrics/gan/test_fid.py'  # requires pytorch_fid
```
```suggestion
            ' --ignore=tests/ignite/metrics/gan/test_fid.py'  # requires pytorch_fid
            ' --ignore=tests/ignite/metrics/nlp/test_rouge.py'  # requires rouge
```

casparvl commented:
Test report by @casparvl
**FAILED**
Build succeeded for 10 out of 11 (4 easyconfigs in total)
gcn6.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/casparvl/984935d816b4a16a2fb245e603195c74 for a full test report.

Micket commented:
My colleague @VRehnberg was working on just filtering out failing tests, but seems like many tests are flaky, and new things are constantly being discovered. I'll update this PR as soon as he's ready.

If anyone is looking to merge some PyTorch goodies there is always the Geometric bundle which should be good to go :)
https://github.com/easybuilders/easybuild-easyconfigs/pull/20013

VRehnberg commented:
> My colleague @VRehnberg was working on just filtering out failing tests, but seems like many tests are flaky, and new things are constantly being discovered. I'll update this PR as soon as he's ready.

Last job looked ok (I'm sure more things will show in some other configuration though) so you could update if you'd like @Micket or I run on some other nodes and see what more shows up first.

Micket commented:
I updated to what you last made, perhaps others want to give it a try as well.

casparvl commented:
I'll give it another go. I had so many failed tests before, it was hard to tell what's what, but I think you excluded many of those, so al least I should (hopefully) get a cleaner log.

casparvl commented:
Test report by @casparvl
**FAILED**
Build succeeded for 4 out of 5 (4 easyconfigs in total)
gcn6.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/casparvl/f133d951fdc1df3dbf45c01668601f62 for a full test report.

Micket commented:
```
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_create_no_dist_gloo - RuntimeError: The server socket has failed to listen on any local network address. The server socket has failed to bind to [::]:2222 (errno: 98 - Address already in use). The server socket has failed to bind to 0.0.0.0:2222 (errno: 98 - Address already in use).
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_create_no_dist_nccl - RuntimeError: The server socket has failed to listen on any local network address. The server socket has failed to bind to [::]:2222 (errno: 98 - Address already in use). The server socket has failed to bind to 0.0.0.0:2222 (errno: 98 - Address already in use).
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_create_dist_gloo_2 - RuntimeError: The server socket has failed to listen on any local network address. The server socket has failed to bind to [::]:2222 (errno: 98 - Address already in use). The server socket has failed to bind to 0.0.0.0:2222 (errno: 98 - Address already in use).
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_create_dist_nccl_2 - RuntimeError: The server socket has failed to listen on any local network address. The server socket has failed to bind to [::]:2222 (errno: 98 - Address already in use). The server socket has failed to bind to 0.0.0.0:2222 (errno: 98 - Address already in use).
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_warning_index_less_localrank - RuntimeError: The server socket has failed to listen on any local network address. The server socket has failed to bind to [::]:2222 (errno: 98 - Address already in use). The server socket has failed to bind to 0.0.0.0:2222 (errno: 98 - Address already in use).
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_spawn_gloo[None] - torch.multiprocessing.spawn.ProcessRaisedException: 
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_spawn_gloo[env://] - torch.multiprocessing.spawn.ProcessRaisedException: 
tests/ignite/distributed/comp_models/test_native.py::test__native_dist_model_spawn_nccl[None] - torch.multiprocessing.spawn.ProcessRaisedException: 
tests/ignite/distributed/utils/test_native.py::test_native_distrib_single_node_spawn_gloo[None] - torch.multiprocessing.spawn.ProcessRaisedException: 
tests/ignite/distributed/utils/test_native.py::test_native_distrib_single_node_spawn_nccl[None] - torch.multiprocessing.spawn.ProcessRaisedException: 
```
I can't inspect the specifics here, but it seems like most failures here are due to port being busy. I could just exclude all of tests/ignite/distributed/ if those are known to be a bit problematic for people when they build.

VRehnberg commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,174 @@
+        'checksums': ['bfe4b6f1cd96e78c021a65a0c51350cdb89d6ef5a8b9609638666ca95bae51d7'],
+        'runtest': (
+            'pytest'
+            ' -m "not multinode_distributed"'
```
```suggestion
            ' -m "not distributed and not multinode_distributed"'
```
> I can't inspect the specifics here, but it seems like most failures here are due to port being busy. I could just exclude all of tests/ignite/distributed/ if those are known to be a bit problematic for people when they build.

This might work in that case.

VRehnberg replied:
Or just ` --ignore tests/ignite/distributed`.

VRehnberg replied:
These I think are from distributed as well:
```
            '" and not test__setup_ddp_vars_from_slurm_env_bad_configs"'  # fails sometimes
            '" and not test__native_dist_model_create_from_backend_bad_config"'  # fails sometimes
```
So could probably removed if covered by a more general rule.

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded (with --ignore-test-failure) for 7 out of 7 (4 easyconfigs in total)
gpu25.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/5a774bd17056eba71c11b16517a39706 for a full test report.

casparvl commented:
Test report by @casparvl
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
gcn6.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/casparvl/4102979959982f9f2a0a575f0dacd647 for a full test report.

VRehnberg commented:
> Test report by @casparvl **FAILED** Build succeeded for 3 out of 4 (4 easyconfigs in total) gcn6.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8 See https://gist.github.com/casparvl/4102979959982f9f2a0a575f0dacd647 for a full test report.

> [31mFAILED[0m tests/ignite/distributed/test_launcher.py::[1mtest_check_idist_parallel_no_dist[0m - subprocess.CalledProcessError: Command '['/scratch-nvme/1/casparl/generic/software/Python/3.11.3...
[31mFAILED[0m tests/ignite/distributed/comp_models/test_native.py::[1mtest__setup_ddp_vars_from_slurm_env_bad_configs[0m - AssertionError: Regex pattern did not match.


Oh, thought these would have been filtered by `-m "not distributed"` but I realise that when I was checking this they where filtered by other reasons. Well, I suppose we should just ignore these specifically again.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 10 out of 12 (4 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/ffeb76e025f1fc493b06f34af92cffff for a full test report.

akesandgren commented on easybuild/easyconfigs/s/SentencePiece/SentencePiece-0.2.0-GCC-12.3.0.eb:
```diff
@@ -0,0 +1,62 @@
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('gperftools', '2.12'),
```
SentencePiece requires protobuf/24.0 as a dependency

Micket replied:
I don't understand how only you managed to trigger this. Also, i assume you meant protobuf-python/24.0

akesandgren replied:
No I mean
```
('protobuf', '24.0')
```

Micket replied:
protobuf-lite is included in sentencepiece as a thirdparty package (maybe we don't actually want that, but regardless), i would have expected it to find that, rather than having the python module link to something potentially different.

> It fails for me since I don't have stuff like that in my container :-)

And surely so does noone else on their OS either. I certainly don't have it, no trace of protobuf, protobuf-lite.


After some digging, I found that one can use external packages for Abseil and protobuf, using
```
'configopts': '-DSPM_PROTOBUF_PROVIDER=package -DSPM_ABSL_PROVIDER=package'
```
when attempting this, I also noticed that 
`'prebuildopts': 'export PKG_CONFIG_PATH=%(installdir)s/lib/pkgconfig:$PKG_CONFIG_PATH`
my pkgconfig stuff ended up in lib64, not lib. Fixing that made `sentencepiece` (the python part) not even look for protobuf. It shouldn't be looking for that stuff to start with, since it should just be using the already compiled `SentencePiece` from the first component. So, I think we have some lib/lib64 difference here. 

I attempted to use external protobuf and Abseil, but I got very very confusing problems. the ABSL_FLAG parsing completely broke, with strange broken output and warnings messages and strange output from even basic things like `spm_train --version`. I couldn't make any sense of it, so I reverted back. 

I've added pkgconf as a builddep and changed the PKG_CONFIG_PATH to include lib64 instead, which I hope will be correct for everyone? I'm not sure. Mixing in a different external protobuf is no good, because it seems to conflict with the bundled thirdparty abseil, and taking in both of them just broke things for me.

akesandgren commented on easybuild/easyconfigs/s/SentencePiece/SentencePiece-0.2.0-GCC-12.3.0.eb:
```diff
@@ -8,9 +8,11 @@ description = "Unsupervised text tokenizer for Neural Network-based text generat
 github_account = 'google'
 
 toolchain = {'name': 'GCC', 'version': '12.3.0'}
+toolchainopts = {'debug': True}
```
Did you really mean to keep this one?

Micket replied:
I did not (leftover from trying to figure out why external ABSL broke it so hard

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/9eb5041ac67af4a9b9c6578c6a08c65e for a full test report.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 5 (5 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/57ab2b534f453a2d86626590b997a036 for a full test report.

jfgrimm commented:
Test report by @jfgrimm
**FAILED**
Build succeeded for 2 out of 6 (5 easyconfigs in total)
gpu24.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/3fa5a56e80cd987ebf7292a4c224ff11 for a full test report.

VRehnberg commented:
> Build fails in torchaudio with:
> 
> ```
>   -- Detecting CXX compile features - done
>   CMake Error at /cvmfs/ebsw.hpc2n.umu.se/amd64_ubuntu2204_zen3/software/CMake/3.26.3-GCCcore-12.3.0/share/cmake-3.26/Modules/CMakeDetermineCUDACompiler.cmake:180 (message):
>     Failed to find nvcc.
> 
>     Compiler requires the CUDA toolkit.  Please set the CUDAToolkit_ROOT
>     variable.
>   Call Stack (most recent call first):
>     CMakeLists.txt:82 (enable_language)
> ```

```
        # Now search for nvcc again using the platform default search paths.
        find_program(_CUDA_NVCC_EXECUTABLE
          NAMES nvcc nvcc.exe
          PATHS ${search_paths}
          PATH_SUFFIXES bin
        )

        # We are done with these variables now, cleanup.
        unset(platform_base)
        unset(possible_paths)
        unset(versions)
        unset(search_paths)

        if(NOT _CUDA_NVCC_EXECUTABLE)
          message(FATAL_ERROR "Failed to find nvcc.\nCompiler ${CMAKE_CUDA_COMPILER_ID} requires the CUDA toolkit. Please set the CUDAToolkit_ROOT variable.")
        endif()
```
Shouldn't `nvcc` be found as long as `CUDA` is loaded? @akesandgren if you do `ml CUDA/12.1.1 && nvcc --version`, does that work?

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Failed during parsing of the easyconfigs, so no ecs were built (1 easyconfigs in total)
alvis1-02 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/c902c5314c53f215d32775a168036eee for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Failed during parsing of the easyconfigs, so no ecs were built (1 easyconfigs in total)
alvis1-03 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/ef981dfd7e538556b00c3d49c7b7dbd1 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-04 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/e98ae38c4f0aa5d708f4b93d4e4c1f70 for a full test report.

akesandgren commented:
One would hope so, but that's what the build tells me, and whether it will be found or not depends on what the CMake file in question actually does.

jfgrimm commented:
@Micket the sentencepiece lib->lib64 change broke my build

Micket commented:
I added both lib/lib64 paths. 

I don't know what to say about the torchaudio problem. I've never seen anything like it, i can't reproduce it so of course I can't even attempt to solve it, so of course I won't even try. Suggestions are welcome.

jfgrimm commented:
@Micket we could try setting `CUDACXX=$EBROOTCUDA/bin/nvcc`

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-03 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/dc401226163ee74b4ec4c13c6c7af847 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-03 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/1c95be580baf4c9e6bc8ef08bd10840e for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-10 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/30dbf126b17ef36b75a952f7e0838b48 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-10 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/a238b75043d22b241f6694a8a956be92 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-03 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/280c5fb32a46a84b4f3ea5cee72479a1 for a full test report.

jfgrimm commented:
Test report by @jfgrimm
**FAILED**
Build succeeded for 2 out of 6 (5 easyconfigs in total)
gpu26.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/02dc5bce6b1af56997abf3e5ecd5d7a1 for a full test report.

Micket commented:
SentencePiece issue I have now boiled down to to it *actually* finding the library like it should, but the sentencepiece.pc file specifies that `protobuf-lite`  is a requirement for the cflags (not actually a requirement for the python wrapper though). But, SentencePiece despite including protocol-buffer as a thirdparty component doesn't actually provide this pc file, so it installs a broken pc file. Ugh. So, one way to use the external protobuf and abseil, which would be a good idea for using this with anything else. After to many hours I dare to admit, I spent half this saturday bonking my head against this, until I learnt that they had simply hacked in a bunch of junk in their bundled ancient Abseil third_party copy.

All those pkg-config things are really just red herrings.

Sheesh. I made some more changes now, I think this builds OK.

Micket commented:
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/1c7fe1dac40792599fcafc00cffc787b for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-06 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/0931d9214e1b42e5f2e5b43661bbab8b for a full test report.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 2 out of 5 (5 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/33ce25bb90f892e7220bbf910fe19a9a for a full test report.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/9fe7ca89ba0eb466a9bfb6c035d8e62b for a full test report.

akesandgren commented on easybuild/easyconfigs/s/SentencePiece/SentencePiece-0.2.0-GCC-12.3.0.eb:
```diff
@@ -0,0 +1,69 @@
+        'easyblock': 'PythonPackage',
+        'start_dir': '%(namelower)s-%(version)s/python',
+        # Unpredicable where pc files end up; including both lib and lib64
+        'prebuildopts': 'export PKG_CONFIG_PATH=%(installdir)s/lib64/pkgconfig:$PKG_CONFIG_PATH && ',
+        'preinstallopts': 'export PKG_CONFIG_PATH=%(installdir)s/lib64/pkgconfig:$PKG_CONFIG_PATH && ',
```
Until https://github.com/easybuilders/easybuild-framework/pull/4435 (i.e. 5.0) you will need to extend PKG_CONFIG_PATH with both lib64 and lib directory versions

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-05 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/905da17c49ec0eb0ad225de050ced7a3 for a full test report.

akesandgren commented:
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/e35a103036f0dba7e24823a26b8d8e88 for a full test report.

Micket commented:
Note: I reverted back to internal absl and protobuf for sentenecepiece as I suspect it the existence of Abseil (or protobuf) caused problems for torch.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 5 (5 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/7122be9584ddb574908908e70ee4783f for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-06 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/c9d58c95b90d804e4ed80a7e440eff4b for a full test report.

akesandgren commented:
Don't merge this yet. There is something odd when I try to build / test it...

akesandgren commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a.eb:
```diff
@@ -0,0 +1,158 @@
+        'checksums': ['7f6b125602ac6d743e523ae0bfa71e1a697a2f5534064528c6ff84c2f7c2fc7f'],
+    }),
+    ('torchvision', '0.16.2', {
+        'preinstallopts': 'WITH_CUDA=1 TORCHVISION_INCLUDE="$EBROOTLIBJPEGMINTURBO/include:$TORCHVISION_INCLUDE"',
```
Drop "WITH_CUDA" here, this is the non-CUDA build

akesandgren commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a.eb:
```diff
@@ -0,0 +1,158 @@
+            'torchaudio-2.2.1_transform_test_tol.patch',
+        ],
+        'preinstallopts': ('rm -r third_party/{sox,ffmpeg/multi};'  # runs twice when testinstall
+                           ' USE_CUDA=1 USE_OPENMP=1 USE_FFMPEG=1 FFMPEG_ROOT="$EBROOTFFMPEG"'),
```
Drop "USE_CUDA" here, still the non-CUDA build

jfgrimm commented:
Test report by @jfgrimm
**FAILED**
Build succeeded for 4 out of 6 (5 easyconfigs in total)
gpu23.viking2.yor.alces.network - Linux Rocky Linux 8.8, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.86.10, Python 3.6.8
See https://gist.github.com/jfgrimm/391d9203484b2521b22de0fa298ca7cb for a full test report.

Micket commented:
@akesandgren just to be clear, those optional things were just excluded (see comments on the GPU build):
- clearml
- mlflow
- neptune
- polyaxon
- tensorboardX
- visdom
- wandb
- pytorch_fid
- rouge
where intentionally just skipped. They aren't actually needed for the builds, and aren't really in any way dependencies, just packages that this bundle can *optionally* interact with. I didn't think it was super necessary to force their inclusion here for everyone just to run a handful of tests. Anyone who wants to use, e.g. wandb, can just load whichever version of that happens to exist in the toolchain independently, and it will still work.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 4 out of 6 (5 easyconfigs in total)
alvis3-25 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, 1 x NVIDIA NVIDIA A100-SXM4-40GB, 545.23.08, Python 3.6.8
See https://gist.github.com/VRehnberg/65872b2869e305e9884226845a28aa2f for a full test report.

Micket commented:
OK, i'm just gonna leave out the CPU version from here, I never actually intended to include it, testing is slow, and evidently the CPU version of PyTorch itself is problematic (and not something I actually care about at all anyway).

VRehnberg commented:
Some nuance

These are different types of "loggers"

>     * clearml
>     * mlflow
>     * neptune
>     * polyaxon
>     * tensorboardX
>     * visdom
>     * wandb

these could be valuable but as they are not part of the PyTorch ecosystem should at least not be extensions of this bundle (but possibly builddeps for the tests). I agree with Micket that

> They aren't actually needed for the builds, and aren't really in any way dependencies, just packages that this bundle can optionally interact with. I didn't think it was super necessary to force their inclusion here for everyone just to run a handful of tests. Anyone who wants to use, e.g. wandb, can just load whichever version of that happens to exist in the toolchain independently, and it will still work.

These:

>     * pytorch_fid 
>     * rouge

are alternative implementations of metrics. They should not be used in combination with Ignite but could be build-dependencies only for the sake of testing. This part holds for these as well:

> They aren't actually needed for the builds, and aren't really in any way dependencies, [...]. I didn't think it was super necessary to force their inclusion here for everyone just to run a handful of tests.

akesandgren commented:
For some reason I have failures in torchaudio on both A100 and V100.
test_dup_hw_acel, test_h264_cuvid and test_hevc_cuvid all fails with "Runtime Error" in torio/io/_streaming_media_decoder.py

Any ideas on why? Or should we just filter them out too?

VRehnberg commented:
We've seen a two errors in the same file, see https://github.com/easybuilders/easybuild-easyconfigs/pull/19987#issuecomment-2013370314

I can't reproduce any other errors in `test/torchaudio_unittest/io/stream_reader_test.py` on 1xV100, 2xV100 or 1xA100. What is the exact error message?

akesandgren commented:
The complete error is:
```
>       self._be.add_video_stream(
            i,
            frames_per_chunk,
            buffer_chunk_size,
            filter_desc,
            decoder,
            decoder_option or {},
            hw_accel,
        )
E       RuntimeError: Failed to initialize CodecContext: Operation not permitted
E
E       To execute this test, run the following from the base repo dir:
E            python test/torchaudio_unittest/io/stream_reader_test.py -k test_h264_cuvid
E
E       This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

/dev/shm/eb_ake/tmp/eb-wv47pwgj/tmp4fweftzb/lib/python3.11/site-packages/torio/io/_streaming_media_decoder.py:822: RuntimeError
```

Micket commented:
My first thought is that it's maybe related to nvenc and permissions to `/render/dri/render*` devices.

akesandgren commented:
Ah yes, we haven't changed permission on /dev/dri/\*render\* on the normal nodes

akesandgren commented:
Esp since they explicitly say "TorchAudio and PyTorch from different releases cannot be used together"

torchdata doesn't have a matrix element for PyTorch 2.1.x so that's going to be a best guess for now.

Micket commented:
yes I did check the compat matrix (and changed the torchvision version compared to the other PRs). I thought I had fixed the torchaudio one, but apparently not. It should be https://github.com/pytorch/audio/releases/tag/v2.1.2

akesandgren commented:
@Micket All three of test_dup_hw_acel, test_h264_cuvid and test_hevc_cuvid fails due to render permission, not just h264

akesandgren commented:
@Micket You forgot to update the torchaudio-2..._use-external-sox.patch for the 2.1.2 version of torchaudio

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-03 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/1b9f3a89aa40a5bfe89408b019c97cdf for a full test report.

Micket commented:
It was actually just me hoping the 2.2.1 patches still applied. Guess not.

VRehnberg commented:
Test report by @VRehnberg
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
alvis1-01 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/e829ac2a8bc66403f62c291e7beb9476 for a full test report.

akesandgren commented:
@Micket You forgot to rename the patch! It's should be 2.1.2 since there will need to be a different one for 2.2.x

akesandgren commented on easybuild/easyconfigs/p/PyTorch-bundle/PyTorch-bundle-2.1.2-foss-2023a-CUDA-12.1.1.eb:
```diff
@@ -0,0 +1,178 @@
+            '" and not kaldi_io_test"'  # requires kaldi_io
+            '" and not test_dup_hw_acel"'  # requires special render device permissions
+            '" and not test_h264_cuvid"'  # requires special render device permissions
+            '" and not test_hecv_cuvid"'  # requires special render device permissions
```
Typo, should be hevc

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-04 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/fac06f4e6ebcdcd4edb03ed44fcd5372 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-04 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/d42b620745b746b59e5a7f49c403dda7 for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-05 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/6b7113318ae567cef3a5ff97d39fdeeb for a full test report.

VRehnberg commented:
Test report by @VRehnberg
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis1-04 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6244 CPU @ 3.60GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 550.54.14, Python 3.6.8
See https://gist.github.com/VRehnberg/e1a80da7762acaa5fa8467499586c470 for a full test report.

Micket commented:
I don't care anymore. I'm just ignoring til this is merged.

akesandgren commented:
Gah.... inject-checksums messed things up...
refixing...

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
b-cn1502.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, 2 x NVIDIA Tesla V100-PCIE-16GB, 545.29.06, Python 3.8.10
See https://gist.github.com/akesandgren/1538a96945938cb2c813ff485d896908 for a full test report.

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/6e4c65774270b117665773d9e099462b for a full test report.

Micket commented:
You also removed the --ignore=tests/ignite/contrib/handlers/test_tqdm_logger.py???

akesandgren commented:
Test report by @akesandgren
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
b-cn1502.hpc2n.umu.se - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, 2 x NVIDIA Tesla V100-PCIE-16GB, 545.29.06, Python 3.8.10
See https://gist.github.com/akesandgren/aefea610bba329c00498ceef17ae8e9d for a full test report.

akesandgren commented:
> You also removed the --ignore=tests/ignite/contrib/handlers/test_tqdm_logger.py???

Yeah, I forgot to reload before adding my stuff...
readded it now.

akesandgren commented:
Test report by @akesandgren
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
b-cn1603.hpc2n.umu.se - Linux Ubuntu 22.04, x86_64, AMD EPYC 7313 16-Core Processor, 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.29.06, Python 3.10.12
See https://gist.github.com/akesandgren/6c4753a4d668255c5a6ae5c2e53c6a48 for a full test report.

akesandgren reviewed (approved):
LGTM

akesandgren commented:
Going in, thanks @Micket!

akesandgren commented:
Going in, thanks @Micket!
